tests/functional/services/auth/test_auth_client_flow.py

```diff
@@ -37,46 +37,50 @@ class CustomAuthClient(globus_sdk.ConfidentialAppAuthClient):
     )
 
 
-# build a partial matrix over
+# build a nearly-diagonal matrix over
 #
 #   domain: str | list[str]
 #   identities: uuid | str | list[uuid] | list[str] | list[str | uuid]
 #   policies: uuid | str | list[uuid] | list[str] | list[str | uuid]
+#   prompt: Literal["prompt"] | None
 #
-# start by seeding a small set of combinations of params to test, then run some loops to
-# fill in the rest
-_ALL_SESSION_PARAM_COMBINATIONS = [
-    (None, None, None),
-    ("foo-id", "bar-id", "baz-id"),
-    (["foo-id", "bar-id"], None, ["quux-id", "snork-id"]),
-    (None, ["foo-id", "bar-id"], ["quux-id", "snork-id"]),
-]
-for domain_option in (None, "example.edu", ["example.edu", "example.org"]):
-    _ALL_SESSION_PARAM_COMBINATIONS.append((domain_option, None, None))
-for identity_option in (
-    None,
+# The order of these options is consequential.
+# They are declared in the same order they tuple-unpack in the parametrized test.
+# `None` values, though valid for each parameter, are omitted;
+# they will be tested exactly once by a single all-`None` option test.
+domain_options = ("example.edu", ["example.edu", "example.org"])
+identity_options = (
     uuid.UUID(int=0),
     "foo-id",
     [uuid.UUID(int=0), uuid.UUID(int=1)],
     ["foo-id", "bar-id"],
     ["foo-id", uuid.UUID(int=2)],
-):
-    _ALL_SESSION_PARAM_COMBINATIONS.append((None, identity_option, None))
-for policy_option in (
-    None,
+)
+policy_options = (
     uuid.UUID(int=3),
     "baz-id",
     [uuid.UUID(int=3), uuid.UUID(int=4)],
     ["baz-id", "quux-id"],
     ["baz-id", uuid.UUID(int=5)],
+)
+prompt_options = ("login",)
+# Seed an all-`None` option test, then use a loop to fill in the rest.
+# There must be 4 parameters so parametrized tuple unpacking works in the test.
+_ALL_SESSION_PARAM_COMBINATIONS = [(None,) * 4]
+for idx, options in enumerate(
+    (domain_options, identity_options, policy_options, prompt_options)
 ):
-    _ALL_SESSION_PARAM_COMBINATIONS.append((None, None, policy_option))
+    for option in options:
+        parameters = [None] * 4
+        parameters[idx] = option
+        _ALL_SESSION_PARAM_COMBINATIONS.append(tuple(parameters))
```

Comment:
Since the authorize URL builder doesn't do any logic to combine any of these, I think this is a very neat solution.

We can always manually append things if there's some combination introduced in the future with special rules/behavior.